datastore/importers/nmdb-import-show-neighbor/Parser.test.cpp

```diff
@@ -0,0 +1,84 @@
+// =============================================================================
+// Copyright 2022 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include <netmeld/datastore/parsers/ParserTestHelper.hpp>
+
+#include "Parser.hpp"
+
+namespace nmdp = netmeld::datastore::parsers;
+
+using qi::ascii::blank;
+
+class TestParser : public Parser {
+    public:
+      using Parser::arpJuniperConfig;
+};
+
+BOOST_AUTO_TEST_CASE(testParts)
```

Comment:
There are a couple of issues here.
1. The `arpJuniperConfig` test logic really is just doing a parser pass/fail and, in that sense, is not really different than the `testWhole` logic.  It's not checking anything about the object(s) that should be created and how.  However, anything else may be more complicated, so see point 2.
2. This should have the other parts as well (and prior to, so as to build up to, the full logic).  Specifically, it should have test logic for `arpHeaderJuniper`, `arpEntryJuniper`, and `arpFooterJuniper`.  If that's there, then it will allow for testing of logic piece-wise and make it easier to check whether a particular logic flow (e.g., whether the "Name" column exists or not) behaves correctly.

Reply:
Let me know if the new tests are good enough to cover these. I tried to order them so it could test for Name or no Name and parse the parts correctly. If there's a better way, let me know.